src/openssl.c

```diff
@@ -3674,6 +3687,8 @@ static int pk_new(lua_State *L) {
 			}
 		}
 
+		ud = prepsimple(L, PKEY_CLASS);
```

Comment:
This can throw and leak `bio` (which is why it was previously done before allocating `bio`)